### DISCUSSION TITLE
cache.invalidate removes all fields for a second

### DISCUSSION THREAD

**Discussion by Miloshinjo** [urql-graphql/urql#2231] [Q&A] [answered]
I have a delete action that I'm doing and I want to do a `cache.invalidate` on the entity that is being deleted in a subscription. 

The entity is nested in another entity (but also I have them standalone sometimes, but they don't cause problems).

When action happens and `cache.invalidate` is fired, all entities in that nested query are removed from UI for a second, and then brought back to UI without the one that is deleted. For 1 or 2 renders the entity arrray is `null`. 

I am wondering why is that happening and is there any known solution to deleting an entity and having it removed from everywhere where it's used, without flashes in UI or sometimes getting `null` for similar entities? :)

Thanks :)

**Comment by JoviDeCroock** [comment, marked as answer]
We can't verify whether the `null` in your array is expected (which is what invalidate does) so we have to refetch that query to ensure we are showing you the correct result.

There are two ways around this
- [manually updating the query](https://formidable.com/open-source/urql/docs/graphcache/cache-updates/#updating-lists-or-links)
- [schema-awareness](https://formidable.com/open-source/urql/docs/graphcache/schema-awareness/) caveat here that the entity should be nullable

**Reply by Miloshinjo** [reply]
Thank you for the quick reply as always. 

Weird, cause we are using schema-awareness and are adding introspected schema (using codegen). Is there any special setting there that should be added?

If it doesn't work I will try and go for the 1st method of manually updating.

**Reply by JoviDeCroock** [reply]
Well what you would have to check is whether that type is nullable or not `[Type!]!` vs `[Type]!` the latter would allow for null the former won't

**Reply by Miloshinjo** [reply]
Thank you for your responses! :)
